File: benchmarks/cpp/integer_boolean/big_integer/GreaterEqual.cpp
#include <iostream>
#include <string>
#include "BUtils.cpp"
#include "BBigInteger.cpp"
#include "BBoolean.cpp"

#ifndef GreaterEqual_H
#define GreaterEqual_H

using namespace std;

class GreaterEqual {



    private:



        BBigInteger counter;

    public:

        GreaterEqual() {
            counter = static_cast<BBigInteger >((BBigInteger("0")));
        }

        void simulate() {
            while((counter.less((BBigInteger("5000000")))._and((BBigInteger("2")).greaterEqual((BBigInteger("1"))))).booleanValue()) {
                counter = static_cast<BBigInteger >(counter.plus((BBigInteger("1"))));
            }
        }

};
int main() {
    clock_t start,finish;
    double time;
    GreaterEqual exec;
    start = clock();
    exec.simulate();
    finish = clock();
    time = (double(finish)-double(start))/CLOCKS_PER_SEC;
    printf("%f\n", time);
    return 0;
}
#endif

File: microbenchmarks/java/integer_boolean/Java_Big_Integer/GreaterEqual.java
import de.hhu.stups.btypes.BInteger;
import de.hhu.stups.btypes.BBoolean;
import de.hhu.stups.btypes.BUtils;

public class GreaterEqual {






    private BInteger counter;

    public GreaterEqual() {
        counter = new BInteger("0");
    }

    public void simulate() {
        while((counter.less(new BInteger("5000000")).and(new BInteger("2").greaterEqual(new BInteger("1")))).booleanValue()) {
            counter = counter.plus(new BInteger("1"));
        }
    }

    public static void main(String[] args) {
        GreaterEqual exec = new GreaterEqual();
        long start = System.nanoTime();
        exec.simulate();
        long end = System.nanoTime();
        System.out.println(exec.getClass().toString() + " Execution: " + (end - start));

        Runtime runtime = Runtime.getRuntime();
        long memory = runtime.totalMemory() - runtime.freeMemory();
        System.out.println("Used memory is bytes: " + memory);
        System.out.println("Used memory is megabytes: "
                + (memory / (1024L * 1024L)));
    }

}
